rbd/watchers_mimic.go

```diff
@@ -8,7 +8,7 @@ package rbd
 #cgo LDFLAGS: -lrbd
 #include <rbd/librbd.h>
 
-extern int callWatchCallback(int index);
+extern void imageWatchCallback(void *);
```

Comment:
Does this need to be a `void *`? It would make it a little clearer if this can be a `uintptr_t` (and `uintptr`).

Reply:
Heh. I'd actually prefer it to be a unitptr_t but cgo won't match the go side of the translation that to the void* that is needed for the ceph callback signature. IMO this was the simplest compromise, at least, t I think that's the case.

Reply:
Also, `imageWatchCallback` is passed directly to `rbd_update_watch()` now, so it must match the callback signature of `rbd_update_callback_t` which is `void (void *)`